src/twisted/web/resource.py

```diff
@@ -72,7 +74,7 @@ def putChild(path, child):
         @type path: C{bytes}
         """
 
-    def render(request):
+    def render(request: IRequest) -> Flattenable:
```

Comment:
Added this type hint without doing the whole file just to ensure that the signatures and types match between `IRenderable` and `Resource`.

Reply:
Must they match though? `twisted.web.resource.Resource` implements `IResource` but does not implement `IRenderable`. They are obviously designed to work together, but when using delegation rather than inheritance, it shouldn't be a problem if `IResource.render()` has a mandatory `request` parameter and `IRenderable.render()` has an optional `request` parameter.